modules/proxy/mod_proxy_ftp.c

```diff
@@ -1176,23 +1126,34 @@ static int proxy_ftp_handler(request_rec *r, proxy_worker *worker,
             }
             return status;
         }
-        /* TODO: see if ftp could use determine_connection */
-        backend->addr = connect_addr;
         ap_set_module_config(c->conn_config, &proxy_ftp_module, backend);
     }
 
+    /*
+     * get all the possible IP addresses for the destname and loop through
+     * them until we get a successful connection
+     */
+    err = ap_proxy_determine_address("FTP", backend, connectname, connectport,
+                                     r, r->server);
+    if (APR_SUCCESS != err) {
+        proxy_ftp_cleanup(r, backend);
```

Comment:
Argh, yes, either that or use NULL instead of r->server in the ap_proxy_determine_address() call to get an APLOG_ERR from there.
I'll go with the latter if my explanation below on why `(r && !s)` is not too twisted :)

Reply:
Fixed in [52126bf](https://github.com/apache/httpd/pull/367/commits/52126bff6facd33ca10f03401565a09aa182018a), using ftp_proxyerror() finally.